src/test/java/org/opentripplanner/middleware/controllers/api/ManageLegTraversalTest.java

```diff
@@ -0,0 +1,306 @@
+package org.opentripplanner.middleware.controllers.api;
+
+import io.leonard.PolylineUtils;
+import io.leonard.Position;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opentripplanner.middleware.models.MonitoredTrip;
+import org.opentripplanner.middleware.models.TrackedJourney;
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.testutils.CommonTestUtils;
+import org.opentripplanner.middleware.testutils.OtpMiddlewareTestEnvironment;
+import org.opentripplanner.middleware.triptracker.ManageLegTraversal;
+import org.opentripplanner.middleware.triptracker.ManageTripTracking;
+import org.opentripplanner.middleware.triptracker.TrackingLocation;
+import org.opentripplanner.middleware.triptracker.TripStatus;
+import org.opentripplanner.middleware.utils.Coordinates;
+import org.opentripplanner.middleware.utils.DateTimeUtils;
+import org.opentripplanner.middleware.utils.JsonUtils;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import static org.opentripplanner.middleware.auth.Auth0Connection.restoreDefaultAuthDisabled;
+import static org.opentripplanner.middleware.auth.Auth0Connection.setAuthDisabled;
+import static org.opentripplanner.middleware.triptracker.ManageLegTraversal.getDistance;
+import static org.opentripplanner.middleware.triptracker.ManageLegTraversal.getSecondsToMilliseconds;
+import static org.opentripplanner.middleware.triptracker.ManageLegTraversal.interpolatePoints;
+
+public class ManageLegTraversalTest extends OtpMiddlewareTestEnvironment {
+
+    private static MonitoredTrip monitoredTrip;
+    private static MonitoredTrip busStopJusticeCenterTrip;
+
+    private static TrackedJourney trackedJourney;
+
+    private static final int SUBWAY_LEG_ID = 1;
+
+    private static final int FINAL_LEG_ID = 2;
+
+    @BeforeAll
+    public static void setUp() throws IOException {
+        assumeTrue(IS_END_TO_END);
```

Comment:
Maybe this is the file that shouldn't require the whole e2e setup. Mongo access is probably not needed. That makes it easier to run everything locally and quickly.

Reply:
No, this is from me initially copying the `TrackedTripControllerTest` test. Have addressed.